src/main.rs

```diff
@@ -1,30 +1,42 @@
 #![allow(clippy::uninlined_format_args)]
 
-use std::fs;
-use std::path::Path;
-
+use clap::builder::{PossibleValuesParser, TypedValueParser};
 use clap::{Parser, Subcommand};
-use tracing::Level;
-use tracing_subscriber::FmtSubscriber;
+use miette::Report;
+use tracing::level_filters::LevelFilter;
 
 mod commands;
 use commands::{Build, Serve};
 
 pub mod message;
-use message::{Message, MessageType};
+use message::OutputFormat;
 
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
 struct Cli {
     #[clap(subcommand)]
     command: Command,
+
+    /// How verbose logging should be (log level)
+    #[clap(long)]
+    #[clap(default_value_t = LevelFilter::WARN)]
+    #[clap(value_parser = PossibleValuesParser::new(["off", "error", "warn", "info", "debug", "trace"]).map(|s| s.parse::<LevelFilter>().expect("possible values are valid")))]
+    #[clap(help_heading = "GLOBAL OPTIONS", global = true)]
+    pub verbose: LevelFilter,
+
+    /// The format of the output
+    #[clap(long, value_enum)]
+    #[clap(default_value_t = OutputFormat::Human)]
+    #[clap(help_heading = "GLOBAL OPTIONS", global = true)]
+    pub output_format: OutputFormat,
 }
 
 #[derive(Subcommand, Debug)]
 enum Command {
     Build(Build),
     Serve(Serve),
 }
+
 fn main() {
     let runtime = tokio::runtime::Builder::new_multi_thread()
         .worker_threads(1)
```

Comment:
it *probably* won't matter in this case but ideally this kind of thing is setup within `run` so that it's all properly shutdown when `run` returns/panics. `start` will call `std::process::exit` on error, so any dtors that are scoped to main will never run.

Reply:
(Cli::parse() is fine/intended to be in main tho)

Reply:
ah word, i mostly just ignored the runtime thing because i know the goal is to eventually get that into axocli- but i agree with your point and will move it before merge